app/src/main/java/org/schabi/newpipe/player/resolver/PlaybackResolver.java

```diff
@@ -67,46 +170,342 @@ default MediaSource buildLiveMediaSource(@NonNull final PlayerDataSource dataSou
+            @NonNull final MediaItemTag metadata) throws IOException {
+        final String url = stream.getContent();
+
+        if (isNullOrEmpty(url)) {
+            throw new IOException(
+                    "Try to generate a progressive media source from an empty string or from a "
+                            + "null object");
+        } else {
+            return dataSource.getProgressiveMediaSourceFactory().createMediaSource(
+                    new MediaItem.Builder()
+                            .setTag(metadata)
+                            .setUri(Uri.parse(url))
+                            .setCustomCacheKey(cacheKey)
+                            .build());
+        }
+    }
+
+    @NonNull
+    private static <T extends Stream> DashMediaSource buildDashMediaSource(
+            @NonNull final PlayerDataSource dataSource,
+            @NonNull final T stream,
+            @NonNull final String cacheKey,
+            @NonNull final MediaItemTag metadata) throws IOException {
+        final boolean isUrlStream = stream.isUrl();
+        if (isUrlStream && isNullOrEmpty(stream.getContent())) {
+            throw new IOException("Try to generate a DASH media source from an empty string or "
+                    + "from a null object");
+        }
+
+        if (isUrlStream) {
+            return dataSource.getDashMediaSourceFactory().createMediaSource(
+                    new MediaItem.Builder()
+                            .setTag(metadata)
+                            .setUri(Uri.parse(stream.getContent()))
+                            .setCustomCacheKey(cacheKey)
+                            .build());
+        } else {
+            String baseUrl = stream.getManifestUrl();
+            if (baseUrl == null) {
+                baseUrl = "";
+            }
+
+            final Uri uri = Uri.parse(baseUrl);
+
+            return dataSource.getDashMediaSourceFactory().createMediaSource(
+                    createDashManifest(stream.getContent(), stream),
+                    new MediaItem.Builder()
+                            .setTag(metadata)
+                            .setUri(uri)
+                            .setCustomCacheKey(cacheKey)
+                            .build());
+        }
+    }
+
+    @NonNull
+    private static <T extends Stream> DashManifest createDashManifest(
+            @NonNull final String manifestContent,
+            @NonNull final T stream) throws IOException {
+        try {
+            final ByteArrayInputStream dashManifestInput = new ByteArrayInputStream(
+                    manifestContent.getBytes(StandardCharsets.UTF_8));
+            String baseUrl = stream.getManifestUrl();
+            if (baseUrl == null) {
+                baseUrl = "";
+            }
+
+            return new DashManifestParser().parse(Uri.parse(baseUrl), dashManifestInput);
+        } catch (final IOException e) {
+            throw new IOException("Error when parsing manual DASH manifest", e);
```

Comment:
And here, there is no need to catch and rethrow with a new error message... Just remove the try catch and let Java propagate the exception normally: the stack trace already tells us that the problem happened while creating the dash manifest, given the method name and the line of the error. Also in all other places.